docs/rationale.md

```diff
@@ -23,4 +23,72 @@ try (Scope scope = TracingContextUtils.currentContextWith(span)) {
+
+
+## Versioning and Releases
+
+### Assumptions
+
+- This project uses semver v2, as does the rest of OpenTelemetry.
+
+### Goals
+
+- API Stability: 
+    - Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, code instrumented with that API module will 
+function, *with no recompilation required*, with any API+SDK that has the same major version, and equal or greater minor or patch version. 
+    - For example, libraries that are instrumented with `opentelemetry-api-trace:1.0.1` will function, at runtime with 
+SDK library `opentelemetry-sdk-trace:1.11.33` plus `opentelemetry-api-trace:1.11.33`. 
+    - We call this requirement the "ABI" compatibility requirement for "Application Binary Interface" compatibility. 
+- SDK Stability:
+    - Public portions of the SDK (constructors, configuration, und-user interfaces) must remain backwards compatible.
+        - Precisely what this includes has yet to be delineated. 
+    - Internal interfaces are allowed to break.
+
+### Methods
+
+- Mature signals
+    - API modules for mature (i.e. released) signals will be transitive dependencies of the `opentelemetry-api` module.
+    - Methods for accessing mature APIs will be added, as appropriate to the `OpenTelemetry` interface.
+    - SDK modules for mature (i.e. released) signals will be transitive dependencies of the `opentelemetry-sdk` module.
+    - Configuration options for the SDK modules for mature signals will be exposed, as appropriate, on the `OpenTelemetrySdk` class.
+    - Modules for these mature signals will be included in the opentelemetry-bom to ensure that users runtime dependencies are kept in sync.
+    - Mixing and matching runtime API and SDK versions, eg. by avoiding use of the BOM, will not be supported by this project.
+    - Once a public API (either in the official API or in the SDK) has been released, we will endeavor to support that API in perpetuity.
+
+- Immature or experimental signals
+    - API modules for immature signals will not be transitive dependencies of the `opentelemetry-api` module.
+    - API modules will be versioned with an "-alpha" suffix to make it abundantly clear that depending on them is at your own risk.
+    - API modules for immature signals will be co-versioned along with mature API modules, with the added suffix.
+    - The java packages for immature APIs will be used as if they were mature signals. This will enable users to easily transition from immature to 
+    mature usage, without having to change imports. 
+    - SDK modules for immature signals will also be versioned with an "-alpha" suffix, in parallel to their API modules.
+    
+### Examples
+
+Purely for illustration purposes, not intended to represent actual releases:
+
+- `v1.0.0` release:
+    - `io.opentelemetry:opentelemetry-api:1.0.0`
+        - Contains APIs for tracing, baggage, propagators (via the context dependency)
```

Comment:
And context itself correct?

Reply:
yes...this was illustrative, not intended to be complete or 100% accurate.

Reply:
added for clarity, though